core/src/main/java/com/github/games647/fastlogin/core/shared/FloodgateManagement.java

```diff
@@ -76,11 +76,38 @@ public void run() {
 
         //this happens on Bukkit if it's connected to Bungee
         //if that's the case, players will be logged in via plugin messages
-        if (core.getStorage() == null) {
+        if (core.getAuthStorage() == null) {
             return;
         }
 
-        profile = core.getStorage().loadProfile(username);
+        profile = core.getAuthStorage().loadProfile(username);
+        if (!profile.isSaved()) {
+            // linked players are stored as Java (= not Floodgate) players
+            profile.setFloodgate(!isLinked);
```

Comment:
Why do you considered this approach? I feel like setting this to `false` could lead to the same issues. Is there any issue leaving it true and then read the configuration instead? This would make it easier to change configuration files.

Maybe we could make this field also a tristate.

Reply:
Linked Floodgate players are 99% Java players: they have the same name (without prefix), the same UUID, inventory, etc. The only difference is that they are present in the Floodgate API and they can't reply to premium java login requests.
I decided to keep it `false`, because they are the same player as their Java part, and I wanted to avoid storing one player as two entries in the database.

Reply:
I've added 5d994d8, but it's still a mess.

Let's say we have a player called Steve, who plays from Java, and linked Floodgate too. What should the value of Floodgate be in our database? Linked or Java? Both of them are correct. Names are unique in the database, so there's no way to have two row for one player.

Reply:
I think I'm lost. Why would we need two rows? You mean for the case Floodgate and Java profiles are associated to different accounts/persons? This would make it very difficult, because the auth plugin has only one single account often based on the player name. However, auth plugins with UUID support would in fact be different. 

The reason I thought name is unique, is because the only a single user can have this Mojang account name. If a user joins with the same name, but a different UUID, this would mean that the old data name is now invalid.

Reply:
I want to keep names unique, changing it would cause a lot of headaches.

What I tried to explain is that Java and their linked Floodgate accounts have the same name (and UUID), and I don't know how to save them in the database.

For example, I join a brand new server from my PC with my Java account. Then I get saved in the database as `Smart123s`, `FloodgateState.JAVA`. Then I get on the bus, grab my phone, and join the server with my linked Floodgate account. My name is still `Smart123s`, because I'm linked, but I'm no longer `FloodgateState.JAVA`, because I'm joining from bedrock this time. Should the database be changed to `Smart123s`, `FloodgateState.LINKED`? And then what happens if I join from Java again?

That's what I initially wanted to avoid by storing linked Floodgate players as Java players.

Reply:
>Should the database be changed to Smart123s, FloodgateState.LINKED? And then what happens if I join from Java again?

Well we could keep it as linked as it would mean (Floodgate+Java), so it doesn't represent the current session, but the account/person that is behind it.

Reply:
Ok, but there are two minor caveats. Firstly, it's impossible to check if a new Java player has a linked Floodgate account until he joins with it. And it's also impossible to detect if a Floodgate account was unlinked. Technically, we could send API requests to Geyser to check for these, but it'd slow down the login process without any actual benefit.

Anyways, it's gonna be a busy week/weekend for me, but I'll check on the code to make sure it works as we discussed above.